sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/PerformanceCounter.cs

```diff
@@ -0,0 +1,181 @@
+using OpenTelemetry.Metrics;
+using System.Diagnostics.Metrics;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal sealed class PerformanceCounter : IDisposable
+    {
+        private readonly MeterProvider _meterProvider;
+        private readonly Meter _perfCounterMeter;
+
+        // Gauge instruments
+        private readonly ObservableGauge<double> _exceptionRateGauge;
+        private readonly ObservableGauge<double> _requestRateGauge;
+        private readonly ObservableGauge<double> _processCpuGauge;
+        private readonly ObservableGauge<double> _processCpuNormalizedGauge;
+        private readonly ObservableGauge<double> _processPrivateBytesGauge;
+        private readonly Process _process = Process.GetCurrentProcess();
+
+        // state for cpu counter calculations
+        private DateTimeOffset _cachedCollectedTime = DateTimeOffset.MinValue;
+        private long _cachedCollectedValueTicks = 0;
+        private double _currentCpuPercentage = 0;
+        private readonly int _processorCount = Environment.ProcessorCount;
+
+        // state for telemetry item related counts
+        private readonly PerfCounterItemCounts _itemCounts;
+        private long _lastExceptionRateCount = 0;
+        private DateTimeOffset _lastExceptionRateTime = DateTimeOffset.UtcNow;
+        private long _lastRequestRateCount = 0;
+        private DateTimeOffset _lastRequestRateTime = DateTimeOffset.UtcNow;
+
+        private bool _disposed;
+
+        public PerformanceCounter(MeterProvider meterProvider, PerfCounterItemCounts itemCounts)
+        {
+            _meterProvider = meterProvider;
+            _perfCounterMeter = new Meter(PerfCounterConstants.PerfCounterMeterName);
+            _itemCounts = itemCounts;
+
+            // Create observable gauges for perf counter
+            _exceptionRateGauge = _perfCounterMeter.CreateObservableGauge(
+                PerfCounterConstants.ExceptionRateInstrumentName,
+                () => GetExceptionRate(),
+                description: "Exception rate gauge (ex/sec)");
+
+            _requestRateGauge = _perfCounterMeter.CreateObservableGauge(
+                PerfCounterConstants.RequestRateInstrumentName,
+                () => GetRequestRate(),
+                description: "Request rate gauge (req/sec)");
+
+            _processCpuGauge = _perfCounterMeter.CreateObservableGauge(
+                PerfCounterConstants.ProcessCpuInstrumentName,
```

Comment:
We should use - https://learn.microsoft.com/en-us/dotnet/core/diagnostics/built-in-metrics-runtime#metric-dotnetprocesscputime